### PR TITLE
Export Data Container in Modal

### PR DESCRIPTION
**Pull request by JCoder121**
First example after clicking the link:
[Deploy Preview](https://deploy-preview-3356--keen-mayer-a86c8b.netlify.app/components/pageheader)

#### What does this PR do?
Adds Export Data modal in the first PageHeader example on the PageHeader components page on the DS site: https://design-system.hpe.design/components/pageheader
according to example + guidance from Figma here: https://www.figma.com/file/BHp0KrOcPvbnzNyshfHkd8/HPE-Page-Header?type=design&node-id=2-3&t=xs7xClQU7yn7WZLM-0

#### Where should the reviewer start?
aries-site/src/examples/components/pageheader/PageHeaderIntroExample.js
aries-site/src/examples/templates/export-data/ExportDataContainer.js

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
First and second ToDos on #3214 

#### Screenshots (if appropriate)
Local environment:
image

#### Should this PR be mentioned in Design System updates?
Yes - when the linked issue ticket is fully closed. After the example on PageHeader gets added, still need to confirm if we should add a dedicated page on the DS site (see the third ToDo)

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible - added a new functionality/example on the PageHeader component page

### CODE REVIEW

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  title,
+}) => {
+  // announce when the layer opens
+  const announce = useContext(AnnounceContext);
+  useEffect(() => {
+    announce(`${title} modal opened.`, 'assertive');
+  }, [announce, title]);
+
+  const [value, setValue] = useState('');
+  const onClose = () => {
```
Comment (halocline) [inline aries-site/src/examples/templates/export-data/ExportDataContainer.js:27]: Move this responsibility to the caller?

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        }}
+      >
+        <Box gap="medium">
+          <ModalBody gap="small">
+            <Paragraph margin="none">Select file type.</Paragraph>
+            <FormField htmlFor="exportData" name="exportData">
+              <RadioButtonGroup
+                id="exportData"
+                name="exportData"
+                options={['CSV', 'JSON', 'PDF']}
```
Comment (halocline) [inline aries-site/src/examples/templates/export-data/ExportDataContainer.js:60]: Let caller specify from `formats` prop and use these as defaults?

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              />
+            </FormField>
+          </ModalBody>
+          <ModalFooter justify="end">
+            <Box direction="row" gap="small">
```
Comment (halocline) [inline aries-site/src/examples/templates/export-data/ExportDataContainer.js:65]: Use ButtonGroup component from aries-core?

**Review by halocline — COMMENTED** [review]
Looking good.

My comments are related to how we might approach this ExportDataContainer as if it were a reusable component. That way if this pattern starts to harden, its something that can "graduate" to a template we might support. Right now, we've started graduating things like ButtonGroup and ModalDialog in this manner.

**Comment by JCoder121** [comment]
> Looking good.
> 
> My comments are related to how we might approach this ExportDataContainer as if it were a reusable component. That way if this pattern starts to harden, its something that can "graduate" to a template we might support. Right now, we've started graduating things like ButtonGroup and ModalDialog in this manner.

Thanks for the feedback! Much of the code was inspired from DestructiveConfirmation.js, and I will implement the changes you mentioned!

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        validate="blur"
+        // eslint-disable-next-line no-unused-vars
+        onSubmit={({ value: formValue, touched }) => {
+          if (onSubmitProp) onSubmitProp(formValue, touched);
+          onClose();
+        }}
+      >
+        <Box gap="medium">
+          <ModalBody gap="small">
+            <Paragraph margin="none">Select file type.</Paragraph>
```
Comment (taysea) [inline aries-site/src/examples/templates/export-data/ExportDataContainer.js:55]: It feels like this Paragraph should sit right under the modal heading as opposed to in the form. Alternately, we could just use the `label` prop on the formfield and say `label="File type"`?

This is more of a design reaction.

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  RadioButtonGroup,
+  Paragraph,
+} from 'grommet';
+import { ModalBody, ModalDialog, ModalFooter } from 'aries-core';
+
+import PropTypes from 'prop-types';
+
+export const ExportDataContainer = ({
+  onSubmit: onSubmitProp,
+  setShowModal,
```
[the remark this answers is not in the document]
Comment (JCoder121) [reply aries-site/src/examples/templates/export-data/ExportDataContainer.js]: I agree! I was wondering for the overall design structure, whether `ExportDataContainer` component should return the selected export format to the caller for other caller logic. I've implemented that type of example within this PR, where I've left template/boilerplate code in the Form's `onSubmit` (see the screenshot) followed by the `onCloseProp`. Let me know what you think regarding this design pattern!
image

File: aries-site/src/examples/templates/export-data/ExportDataContainer.js (hunk 4 above)
Comment (JCoder121) [reply aries-site/src/examples/templates/export-data/ExportDataContainer.js]: Landed on using FormField's `label=File type`

**Review by vavalos5 — APPROVED** [review]
Thank you for doing this, looks great!

File: aries-site/src/examples/components/pageheader/PageHeaderIntroExample.js [hunk 6]
```diff
@@ -1,17 +1,41 @@
...
+
 import { ReverseAnchor } from '../../templates';
 
-const actions = [
-  {
-    label: 'Export devices',
-    secondary: true,
-  },
-];
+const label = 'Export Report';
```
Comment (halocline) [inline aries-site/src/examples/components/pageheader/PageHeaderIntroExample.js:15]: Sentence case

```suggestion
const label = 'Export report';
```

**Review by halocline — APPROVED** [review]

**Comment by halocline** [comment]
Thank you @JCoder121 !! You rock.

**Pull request merged**